### PR TITLE
test: restore unit test coverage to 95% threshold

### LINKED ISSUE #5564: Restore code coverage to 95%
Upgrading to [vitest v4](https://vitest.dev/blog/vitest-4) in #5561 reduced our code coverage percentage because the coverage calculation changed. To upgrade the dependency, our code coverage thresholds were slightly reduced (93/89/93/90% instead of 95%). We should add a few more tests so that we can increase the threshold back to 95%.

### PR DESCRIPTION
**Pull request by spencer-brooks-sf**
## Details

Resolves #5564.

The vitest v4 upgrade (#5561) changed the coverage calculation, which dropped our percentages below the previous 95% threshold. The thresholds were temporarily lowered to unblock the upgrade.

This PR adds unit tests across multiple packages to restore coverage to 95% across all four metrics (branches, functions, lines, statements), then raises the thresholds back:

- `@lwc/compiler` — options validation edge cases
- `@lwc/style-compiler` — CSS transform error paths
- `@lwc/engine-server` — SSR renderer branches (render modes, slot fallback, inner/outer HTML, static content)
- `@lwc/features` — setFeatureFlag validation
- `@lwc/shared` — api-version ranges, assert, normalize-class, overridable-hooks
- `@lwc/ssr-compiler` — compilation errors, estree builders, transmogrify, template context
- `@lwc/ssr-runtime` — class-list, context, lightning-element, to-iterator-directive, wire
- `@lwc/template-compiler` — collect-params edge cases
- `@lwc/wire-service` — registration, connection, config updates

Also updates `vitest.config.mjs` to restore 95% thresholds and removes the TODO comment referencing this issue.

## Does this pull request introduce a breaking change?

- 😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

- 🤞 No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts [hunk 1]
```diff
@@ -143,6 +143,32 @@ describe('context', () => {
             process.env.NODE_ENV = originalNodeEnv;
         });
 
+        it('silently ignores duplicate provideContext calls in production', () => {
+            vi.stubEnv('NODE_ENV', 'production');
```
Comment (lpomerleau) [inline packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts]: This test uses `vi.stubEnv`,  while the surrounding describe block uses direct `process.env.NODE_ENV = ...` assignment with a shared originalNodeEnv. Should we use 1 style to avoid globals clobbering each other?

File: packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const el = new LightningElement({ tagName: 'x-test' });
+    el[SYMBOL__SET_INTERNALS]({}, {}, new Set());
+    return el;
+}
+
+describe('wire', () => {
+    beforeEach(() => {
+        if (!(globalThis as any).lwcRuntimeFlags) {
+            (globalThis as any).lwcRuntimeFlags = {};
+        }
```
Comment (lpomerleau) [inline packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts]: Why only set if it's empty? Does it need cleanup in `afterEach`?

File: packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts (hunk 1 above)
Comment (spencer-brooks-sf) [reply packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts]: Yeah good call process.env usage is what has traditionally been done, i can just shift to that since its more wide spread usage.

File: packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts (hunk 2 above)
Comment (spencer-brooks-sf) [reply packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts]: It was to avoid clobbering it if something else has set it that should roughly persisted, but could also just be something always set and cleaned up?

Comment (lpomerleau) [reply packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts]: I'd lean toward always setting it then cleaning it up. Other code outside this file should be independent. Do the tests pass if the runtime flags are always empty here?

File: packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { createContextProvider } from '../wire';
+import { LightningElement, SYMBOL__SET_INTERNALS } from '../lightning-element';
+import type { WireAdapterConstructor } from '@lwc/engine-core';
+
+class FakeAdapter {
+    update() {}
+    connect() {}
+    disconnect() {}
+}
+const fakeAdapter: WireAdapterConstructor = FakeAdapter;
```
Comment (wjhsf) [inline packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts]:
```suggestion
```
This isn't necessary.

File: packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+describe('wire', () => {
+    beforeEach(() => {
+        (globalThis as any).lwcRuntimeFlags = {};
+    });
+
+    afterEach(() => {
+        delete (globalThis as any).lwcRuntimeFlags;
+    });
```
Comment (wjhsf) [inline packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts:31]:
```suggestion
```
If we're just setting `lwcRuntimeFlags` to empty, then we shouldn't need to set it at all.

File: packages/@lwc/ssr-runtime/src/__tests__/lightning-element.spec.ts [hunk 5]
```diff
@@ -95,4 +95,21 @@ describe('SSR LightningElement', () => {
             expect(el.template).toEqual({ synthetic: false });
         });
     });
+
+    describe('hidden reflection', () => {
```
Comment (wjhsf) [inline packages/@lwc/ssr-runtime/src/__tests__/lightning-element.spec.ts]: `describe('hidden reflection')` feels like poetry about a zen garden

File: packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts [hunk 6]
```diff
@@ -143,6 +143,33 @@ describe('context', () => {
             process.env.NODE_ENV = originalNodeEnv;
         });
 
+        it('silently ignores duplicate provideContext calls in production', () => {
+            const originalNodeEnv = process.env.NODE_ENV;
+            process.env.NODE_ENV = 'production';
+            try {
```
Comment (wjhsf) [inline packages/@lwc/ssr-runtime/src/__tests__/context.spec.ts]: This is weird, we should use standard before/after blocks.

File: packages/@lwc/ssr-compiler/src/__tests__/compilation.spec.ts [hunk 7]
```diff
@@ -135,6 +135,110 @@ export default class Test extends LightningElement {
                   }
                 `);
         });
+
+        test('throws when @wire is called with no arguments', () => {
+            const src = `import { wire, LightningElement } from "lwc";
```
Comment (wjhsf) [inline packages/@lwc/ssr-compiler/src/__tests__/compilation.spec.ts]: These should be fixture tests, rather than inline code. See the `fixtures` directory for other examples.

Comment (wjhsf) [inline packages/@lwc/ssr-compiler/src/__tests__/compilation.spec.ts]: These should be fixture tests, rather than inline code. See the `fixtures` directory for other examples.

File: packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts (hunk 4 above)
Comment (spencer-brooks-sf) [reply packages/@lwc/ssr-runtime/src/__tests__/wire.spec.ts:31]: Will throw an error if its not at least an object. Theres runtime code that is being test that requires lwcRuntimeFlags to at least be an object even if it contains no flags inside it.

**Review by wjhsf — APPROVED** [review]
